google/cloud/bigtable/tests/data_integration_test.cc

```diff
@@ -673,6 +676,61 @@ TEST(ConnectionRefresh, Frequent) {
   Apply(table, row_key, created);
 }
 
+TEST_P(DataIntegrationTest, ClientQueryTest) {
+  auto const table_id = testing::TableTestEnvironment::table_id();
+  auto table = Table(
+      MakeDataConnection(
+          Options{}
+              .set<DataRetryPolicyOption>(
+                  DataLimitedErrorCountRetryPolicy(0).clone())
+              .set<DataBackoffPolicyOption>(
+                  google::cloud::internal::ExponentialBackoffPolicy(ms(0),
+                                                                    ms(0), 2.0)
+                      .clone())
+              .set<bigtable::experimental::QueryPlanRefreshRetryPolicyOption>(
+                  bigtable::experimental::
+                      QueryPlanRefreshLimitedErrorCountRetryPolicy(0)
+                          .clone())),
+      TableResource(project_id(), instance_id(), table_id));
+  std::string const row_key = "row-key-for-client-query-test";
+  std::string const family = kFamily4;
+  std::string const column1 = "c1";
+  std::string const column2 = "c2";
+  std::string const value1 = "v1";
+  std::string const value2 = "v2";
+
+  std::vector<Cell> created{
+      {row_key, family, column1, 0, value1},
+      {row_key, family, column2, 0, value2},
+  };
+  BulkApply(table, created);
+  auto client = Client(table.connection());
+  std::vector<std::string> full_table_path =
+      absl::StrSplit(table.table_name(), '/');
+  auto table_name = full_table_path.back();
+  std::string quoted_table_name = "`" + table_name + "`";
+  Project project(project_id());
+  InstanceResource instance_resource(project, instance_id());
+  auto prepared_query = client.PrepareQuery(
+      instance_resource,
+      SqlStatement("SELECT CAST(family4['c0'] AS STRING) AS c0  FROM " +
+                   quoted_table_name + " WHERE _key = '" + row_key + "'"));
+  ASSERT_STATUS_OK(prepared_query);
+
+  auto bound_query = prepared_query->BindParameters({});
+  auto row_stream = client.ExecuteQuery(std::move(bound_query));
+
+  std::vector<StatusOr<bigtable::QueryRow>> rows;
+  for (auto const& row : std::move(row_stream)) {
+    rows.push_back(row);
+  }
+
+  ASSERT_EQ(rows.size(), 1);
+  ASSERT_STATUS_OK(rows[0]);
+  auto const& row1 = *rows[0];
+  EXPECT_THAT(row1.columns().at(0), "c0");
```

Comment:
This is probably an assert eq, right? (I'm checking your code to come up with my tests)

Reply:
Yup, thanks. Fixed these assertions.